File: MoveToNext8th.js
var SCRIPT_TITLE = "Move selected note 8th note later";

function getClientInfo() {
  return {
    "name": SV.T(SCRIPT_TITLE),
    "author": "lag129",
    "versionNumber": 1,
    "minEditorVersion": 0
  }
}

function getTranslations(langCode) {
  if(langCode == "ja-jp") {
    return [
      [SCRIPT_TITLE, "8分後に選択ノートを移動する"]
    ];
  }
  return [];
}

function main() {
    var selection = SV.getMainEditor().getSelection();
    var selectedNotes = selection.getSelectedNotes();

    if(selectedNotes.length == 0){
        SV.finish();
    }

    for(var i=0; i<selectedNotes.length; i++){
		
		var timeAxis = SV.getProject().getTimeAxis();
		var tempoMark = timeAxis.getTempoMarkAt(currentOnset);

		var currentOnset = selectedNotes[i].getOnset();
		var bpm = tempoMark.bpm;
		var currentBar = ((60/bpm)*4)/8
		var currentBarBlick = timeAxis.getBlickFromSeconds(currentBar)
		var diffBlick = currentOnset + currentBarBlick

		selectedNotes[i].setOnset(diffBlick);
    }

  SV.finish();
}

File: MoveToNext8th.lua
SCRIPT_TITLE = "Move selected note 8th note later"

function getClientInfo()
  return {
    name = SV:T(SCRIPT_TITLE),
    author = "lag129",
    versionNumber = 1,
    minEditorVersion = 68354
  }
end

function getTranslations(langCode)
  if langCode == "ja-jp" then
    return {
      { SCRIPT_TITLE, "8分後に選択ノートを移動する" }
    }
  end
  return {}
end

function main()
  local selection = SV:getMainEditor():getSelection()
  local selectedNotes = selection:getSelectedNotes()

  if #selectedNotes == 0 then
    SV:finish()
    return
  end

  for i = 1, #selectedNotes do
    local note = selectedNotes[i]
    local currentOnset = note:getOnset()
    local timeAxis = SV:getProject():getTimeAxis()
    local tempoMark = timeAxis:getTempoMarkAt(currentOnset)

    local bpm = tonumber(tempoMark.bpm)
    local currentBar = ((60 / bpm) * 4) / 8
    local currentBarBlick = timeAxis:getBlickFromSeconds(currentBar)
    local diffBlick = currentOnset + currentBarBlick

    selectedNotes[i]:setOnset(diffBlick)
  end

  SV:finish()
end
